js.go

```diff
@@ -1180,37 +1292,44 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 					return nil, ErrSubjectMismatch
 				}
 
+				// Update attached status.
```

Comment:
What if you add a LABEL before processing of `switch{}` where we have `case info != nil`, then line 1288 (after calling js.ConsumerInfo() when return is OK, we could here simply make sure that the subscription is destroyed, 
set `deliver` to info.Config.DeliverSubject (although likely set in the processing of the info in switch statement), set `shouldCreate` to false, so we don't attempt again to add the consumer, then a simple goto <LABEL>. That should take care of validation and recreating the sub/jsi from scratch.

note1: recreate also the `ch` for sync subscriptions because it has been assigned to another subscription, so you can't reused. This is an issue I found in the past.

note2: you would also have to move the declaration of `ccreq *createConsumerRequest` and some other at the top of the function).

That's how I do in the C client and find it a bit more clean. That being said, this is with my pull subscribe implementation that uses a single sync subscribe, so basically I have only a single js.nc.subscribe() call, it's just that the deliver subject is an inbox in case of pull subscribe (and a sync flag).

Reply:
Code is working, so rather not redo completely again.